src/main/java/org/jabref/gui/exporter/ExportToClipboardAction.java

```diff
@@ -55,48 +51,45 @@ public void run() {
             return;
         }
 
-        List<Exporter> exporters = Globals.exportFactory.getExporters();
-        exporters.sort(Comparator.comparing(Exporter::getDisplayName));
-        List<String> exportFormatDisplayNames = exporters.stream().map(Exporter::getDisplayName).collect(Collectors.toList());
-
-        JList<String> list = new JList<>(exportFormatDisplayNames.toArray(new String[exportFormatDisplayNames.size()]));
-        list.setBorder(BorderFactory.createEtchedBorder());
-        list.setSelectionInterval(0, 0);
-        list.setSelectionMode(ListSelectionModel.SINGLE_SELECTION);
-        int answer = JOptionPane.showOptionDialog(null, list, Localization.lang("Select export format"),
-                JOptionPane.YES_NO_OPTION, JOptionPane.QUESTION_MESSAGE, null,
-                new String[] {Localization.lang("Export"),
-                        Localization.lang("Cancel")},
-                Localization.lang("Export"));
-        if (answer == JOptionPane.NO_OPTION) {
+        List<Exporter> exporters = new ArrayList<>(Globals.exportFactory.getExporters());
+
+        Optional<Exporter> selectedExporter = frame.getDialogService().showChoiceDialogAndWait(Localization.lang("Export"), Localization.lang("Select export format"),
+                Localization.lang("Export"), null, exporters);
+
+        if (!selectedExporter.isPresent()) {
             return;
         }
 
-        Exporter exporter = exporters.get(list.getSelectedIndex());
+        Exporter exporter = selectedExporter.get();
 
         // Set the global variable for this database's file directory before exporting,
         // so formatters can resolve linked files correctly.
         // (This is an ugly hack!)
-        Globals.prefs.fileDirForDatabase = frame.getCurrentBasePanel().getBibDatabaseContext()
+        Globals.prefs.fileDirForDatabase = frame.getCurrentBasePanel()
+                .getBibDatabaseContext()
                 .getFileDirectories(Globals.prefs.getFileDirectoryPreferences());
 
-        File tmp = null;
+        Path tmp = null;
         try {
             // To simplify the exporter API we simply do a normal export to a temporary
             // file, and read the contents afterwards:
-            tmp = File.createTempFile("jabrefCb", ".tmp");
-            tmp.deleteOnExit();
+            tmp = Files.createTempFile("jabrefCb", ".tmp");
+
             List<BibEntry> entries = panel.getSelectedEntries();
 
             // Write to file:
-            exporter.export(panel.getBibDatabaseContext(), tmp.toPath(),
-                    panel.getBibDatabaseContext().getMetaData().getEncoding()
+            exporter.export(panel.getBibDatabaseContext(), tmp,
+                    panel.getBibDatabaseContext()
```

Comment:
Sorry for being pedantic, but please fix the code style (make it less aggressive and align points). Otherwise the code gets reformated a lot and actually in a "wrong" format.

Reply:
Do you have two lines or one line indentation as default?

![grafik](https://user-images.githubusercontent.com/320228/37120658-48a7ae36-225b-11e8-873b-58f5900d47df.png)